### ISSUE TITLE
Argument of type { [x: string]: Blob; } is not assignable to parameter of type Record<string, ClipboardItemData>

### ISSUE THREAD

**Issue by Bahman-66** [microsoft/TypeScript#46116]
# lib Update Request

## Configuration Check

My compilation *target* is `ES2017` and my *lib* is `ES2018`.

## Missing / Incorrect Definition

    error TS2345: Argument of type '{ [x: string]: Blob; }' is not assignable to parameter of type 'Record<string, ClipboardItemData>'.
      'string' index signatures are incompatible.
        Type 'Blob' is missing the following properties from type 'Promise<ClipboardItemDataType>': then, catch, finally, [Symbol.toStringTag]
    

## Sample Code

 
    graphicDiv.addEventListener(
      'dragend',
      (event) => {
        const canvas = <HTMLCanvasElement>document.getElementById('canvas');
        canvas.toBlob((blob) => {
          if (blob) {
            let data = [new ClipboardItem({ [blob.type]: blob })];

            navigator.clipboard.write(data).then(
              function () {},
              function (err) {}
            );
          }
        });
      },
      false
    );   

## Documentation Link

https://developer.mozilla.org/en-US/docs/Web/API/Navigator/clipboard

**Comment by Bahman-66** [comment]
I manipulated the lib file so ClipboardItem constructor accepted any types for items and also the Clipbard.write method accept any parameter the results is as follow:

linux machine

- chrome 

`TypeError: Failed to construct 'ClipboardItem': Failed to convert value to 'Blob'.`
 

- firefox 

`ReferenceError: ClipboardItem is not defined`

-edge beta version
`successfully copied the image from blob to clipboard`

windows machine 

 all of them fails similar to firefox and chrome on linux

**Comment by andrewbranch** [comment]
Can you show exactly what change you are proposing to make to what definition? I’m finding this kind of hard to follow without going and looking up a bunch of stuff.

**Comment by IllusionMH** [comment]
I think changes are already in place: [error in 4.4.3](https://www.typescriptlang.org/play?ts=4.4.3#code/CYUwxgNghgTiAEYD2A7AzgF3gIwFzwCEIlsBuAWACgqUQB3eAYQgEsAHbJWYASQxAC2ACgDeAchYCoAcxAB6Nimlj82AL4BKUkA) and [works in Nightly](https://www.typescriptlang.org/play?ts=4.5.0-dev.20211001#code/CYUwxgNghgTiAEYD2A7AzgF3gIwFzwCEIlsBuAWACgqUQB3eAYQgEsAHbJWYASQxAC2ACgDeAchYCoAcxAB6Nimlj82AL4BKUkA)
Previously argument was
```ts
Record<string, ClipboardItemData>
```
but in Nightly it's
```ts
Record<string, ClipboardItemDataType | PromiseLike<ClipboardItemDataType>>
```

**Comment by IllusionMH** [comment]
It was fixed in https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1102

And there is further PR to leave only `Blob` at this moment https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1139

I believe this issue can be closed (but I'm not reporter).

**Comment by andrewbranch** [comment]
Thanks @IllusionMH!
